akka-cluster-sharding/src/main/scala/akka/cluster/sharding/internal/EventSourcedRememberShards.scala

```diff
@@ -16,33 +21,60 @@ import akka.persistence.PersistentActor
 private[akka] object EventSourcedRememberShards {
   def props(typeName: String): Props =
     Props(new EventSourcedRememberShards(typeName))
+
+  class FromOldCoordinatorState() extends EventAdapter {
+    override def manifest(event: Any): String =
+      ""
+
+    override def toJournal(event: Any): Any =
+      event
+
+    override def fromJournal(event: Any, manifest: String): EventSeq = {
+      event match {
+        case ShardHomeAllocated(shardId, _) =>
+          EventSeq.single(shardId)
+        case _ => EventSeq.empty
+      }
+
+    }
+  }
 }
 
 /**
  * INTERNAL API
+ *
+ * FIXME add snapshotting and then the migration event adapter can be removed
  */
 @InternalApi
-private[akka] final class EventSourcedRememberShards(typeName: String) extends PersistentActor {
+private[akka] final class EventSourcedRememberShards(typeName: String) extends PersistentActor with ActorLogging {
 
-  override val persistenceId: String = s"$typeName-remember-entitites"
+  // Uses the same persistence id as the old persistent coordinator so that the old data can be migrated
+  // without any user action
+  override def persistenceId = s"/sharding/${typeName}Coordinator"
 
-  private var shards = Set.empty[ShardId]
+  private val shards = mutable.Set.empty[ShardId]
 
   override def receiveRecover: Receive = {
     case shardId: ShardId =>
-      // FIXME optimize for adding rather than reading (which is done only once)
-      shards += shardId
+      shards.add(shardId)
+    case SnapshotOffer(_, state: Internal.State) =>
+      shards ++= (state.shards.keys ++ state.unallocatedShards)
```

Comment:
I guess we should have/add our own snapshot as well?

Reply:
Yep snapshot is still a TODO